### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by ChetCorey**
For the most part, the primary application has not been altered. I made some small changes to the LeftMenu and how the data is stored. Changing it from a collection of values to an observable object.

### CODE REVIEW

File: app/scripts/components/dashboardComponents/breadcrumbs.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                        onClick={() => dashboardStore.toggleCollapseTree(this.props.router)}
+                    />
+                    {this.breadCrumb()}
+                </Paper> 
+          	</div>
+        );
+    }
+
+    pathFinder(kind) {
+        let kinds = {
```
Comment (caseychoiniere) [inline app/scripts/components/dashboardComponents/breadcrumbs.jsx]: There's an enum for kinds in `util/urlEnum.js`. That way if an api change is made we can just change the kind string in one place.

File: app/scripts/components/dashboardComponents/breadcrumbs.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import React, { Component, PropTypes } from 'react'
```
Comment (caseychoiniere) [inline app/scripts/components/dashboardComponents/breadcrumbs.jsx]: You'll want to remove the PropTypes import from React. The package is now separate and it looks like you have both listed here.

File: app/scripts/components/dashboardComponents/treeList.jsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
+import React, { Component, PropTypes } from 'react';
```
Comment (caseychoiniere) [inline app/scripts/components/dashboardComponents/treeList.jsx]: remove PropTypes from React and just use `const {} = PropTypes`

File: app/scripts/components/dashboardComponents/treeList.jsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            e.preventDefault();
+            this.handleTouchTap(item)
+        }
+    }
+    
+    handleTouchTap(item) {
+        dashboardStore.selectItem(item.id);
+    }
+
+    iconPicker(child, ancestorIds) {
```
Comment (caseychoiniere) [inline app/scripts/components/dashboardComponents/treeList.jsx:82]: consider using values in `urlEnum.js`

File: app/scripts/components/globalComponents/leftMenu.jsx [hunk 5]
```diff
@@ -138,10 +138,13 @@ class LeftMenu extends React.Component {
     }
 }
 
+const toolbarHeight = '56px'
```
Comment (caseychoiniere) [inline app/scripts/components/globalComponents/leftMenu.jsx]: Why are you setting this variable and then using it in your styles object? Why not just set `top: 56` and `height: 56`?

File: app/scripts/stores/dashboardStore.js [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+                }
+                mainStore.listItems = parentsChildren
+                this.responseHeaders = headers;
+                mainStore.loading = false;
+            }).catch(ex =>this.handleErrors(ex))
+    }
+
+    @action toggleDrawer() {
+        let drawerPosition = !this.drawer.get('open')
+        let contentStyle = { transition: 'margin-left 450ms cubic-bezier(0.23, 1, 0.32, 1)' };
```
Comment (caseychoiniere) [inline app/scripts/stores/dashboardStore.js]: move this out of store? Keep stores to logic only as much as possible.

File: app/scripts/stores/dashboardStore.js [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+        drawerPosition ? contentStyle.marginLeft = this.drawer.get('width') : null
+        this.drawer.set('open', drawerPosition)
+        this.drawer.set('contentStyle', contentStyle)
+    }
+    
+    @action setDrawer() {
+        let width = 350;
```
Comment (caseychoiniere) [inline app/scripts/stores/dashboardStore.js]: move styles to component

File: app/scripts/stores/dashboardStore.js [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+                        json.open = true
+                        json.childrenIds = item.childrenIds
+                        json.folderIds = item.folderIds
+                        this.downloadedItems.delete(id)
+                    }
+                    this.downloadedItems.set(id, json)
+                    if(isSelected) this.selectedItem = id
+                    mainStore.loading = false;
+                } else {
+                    json.code === 'resource_not_consistent' ? mainStore.tryAsyncAgain(getItem, [id, path, isSelected]) : mainStore.handleErrors(json);
```
Comment (caseychoiniere) [inline app/scripts/stores/dashboardStore.js:227]: I think that you're missing some params for `tryAsyncAgain()`

Should have `tryAsyncAgain(func, args, delay, counterId, message, isUpload)`

1.) function to retry
2.) arguments array that the function to be retried needs
3.) A delay in milliseconds
4.)An ID for the counter (that way each counter is unique) you can just use the `id`of the item
5.) A message to display to the user
6.) whether or not it's an upload (bool)

File: app/scripts/stores/dashboardStore.js [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+                this.selectedItem = id
+            } else {
+                this.getItem(id, `${path}/`, true)
+            }
+        }
+    }
+
+    @action selectItem(itemId) {
+        this.drawer.set('toggleLable', 'Home');
+        mainStore.filesChecked = []
```
Comment (caseychoiniere) [inline app/scripts/stores/dashboardStore.js:245]: I actually want to refactor how this is done in the mainStore. Not necessary for this pull request though. I don't like how I used arrays of file or folder id's when items are selected in the listItems.jsx component. Not sure why I did it this way before. Should be using an observable.map() or even just an array of selected items where each item in the array is the file or folder object with all properties. Then filter by kind as needed. I will refactor this after we pull in your changes.

File: app/scripts/stores/mainStore.js [hunk 10]
```diff
@@ -231,18 +231,23 @@ export class MainStore {
...
 
     @action setLeftNavIndex(index) {
-        this.leftNavIndex = index;
+        this.leftMenuDrawer.set('index', index);
     }
 
     @action toggleNavDrawer() {
-        this.toggleNav = !this.toggleNav;
+        let drawerOpen = this.leftMenuDrawer.get('open')
+        this.leftMenuDrawer.set('open', !drawerOpen)
```
Comment (caseychoiniere) [inline app/scripts/stores/mainStore.js]: maybe???   `this.leftMenuDrawer.set('open', !this.leftMenuDrawer.get('open'))`

File: app/scripts/stores/mainStore.js [hunk 11]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
-import { observable, computed, action, map } from 'mobx';
+import { observable, computed, action, map, extendObservable } from 'mobx';
```
Comment (caseychoiniere) [inline app/scripts/stores/mainStore.js]: Are you ever using extendObservable? If not, remove this.

**Review by caseychoiniere — CHANGES_REQUESTED** [review]
Added some comments

File: app/scripts/stores/mainStore.js (hunk 10 above)
Comment (ChetCorey) [reply app/scripts/stores/mainStore.js]: Yep, that's just as clear if not more and less code 👍

File: app/scripts/components/dashboardComponents/breadcrumbs.jsx (hunk 1 above)
Comment (ChetCorey) [reply app/scripts/components/dashboardComponents/breadcrumbs.jsx]: Changed it to a case switch so I could use Kind. I just pushed up those changes.

File: app/scripts/components/dashboardComponents/breadcrumbs.jsx (hunk 2 above)
Comment (ChetCorey) [reply app/scripts/components/dashboardComponents/breadcrumbs.jsx]: Mostly Notes for myself
Push up fixes for this

File: app/scripts/components/dashboardComponents/treeList.jsx (hunk 3 above)
Comment (ChetCorey) [reply app/scripts/components/dashboardComponents/treeList.jsx]: Mostly Notes for myself
Push up fixes for this

File: app/scripts/components/globalComponents/leftMenu.jsx (hunk 5 above)
Comment (ChetCorey) [reply app/scripts/components/globalComponents/leftMenu.jsx]: Mostly Notes for myself
Push up fixes for this

File: app/scripts/stores/mainStore.js (hunk 10 above)
Comment (ChetCorey) [reply app/scripts/stores/mainStore.js]: Mostly Notes for myself
Push up fixes for this

File: app/scripts/stores/mainStore.js (hunk 11 above)
Comment (ChetCorey) [reply app/scripts/stores/mainStore.js]: Mostly Notes for myself
Push up fixes for this

File: app/scripts/components/dashboardComponents/treeList.jsx (hunk 4 above)
Comment (ChetCorey) [reply app/scripts/components/dashboardComponents/treeList.jsx:82]: I have notes to Work on this on Monday

File: app/scripts/stores/dashboardStore.js (hunk 6 above)
Comment (ChetCorey) [reply app/scripts/stores/dashboardStore.js]: I will work on this on Monday

File: app/scripts/stores/dashboardStore.js (hunk 7 above)
Comment (ChetCorey) [reply app/scripts/stores/dashboardStore.js]: I will work on this on Monday

File: app/scripts/stores/dashboardStore.js (hunk 8 above)
Comment (ChetCorey) [reply app/scripts/stores/dashboardStore.js:227]: These are great notes, Thanks
I will work on this on Monday

File: app/scripts/components/dashboardComponents/treeList.jsx (hunk 4 above)
Comment (ChetCorey) [reply app/scripts/components/dashboardComponents/treeList.jsx:82]: Fixed, Committed and Pushed

**Review by caseychoiniere — APPROVED** [review]

File: app/scripts/stores/dashboardStore.js (hunk 8 above)
Comment (caseychoiniere) [reply app/scripts/stores/dashboardStore.js:227]: I approved this PR but just realized that these changes still need to be made.

File: app/scripts/stores/dashboardStore.js (hunk 7 above)
Comment (caseychoiniere) [reply app/scripts/stores/dashboardStore.js]: I approved this PR but just realized that these changes still need to be made.

**Review by caseychoiniere — CHANGES_REQUESTED** [review]

**Review by caseychoiniere — APPROVED** [review]

**Comment by caseychoiniere** [comment]
+1

File: app/scripts/stores/mainStore.js [hunk 12]
```diff
@@ -195,7 +197,6 @@ export class MainStore {
         this.templateProperties = [];
         this.toasts = [];
         this.totalItems = null;
-        this.toggleNav = false;
```
Comment (caseychoiniere) [inline app/scripts/stores/mainStore.js:198]: I'm guessing that this will break a test.

File: app/scripts/stores/mainStore.js [hunk 13]
```diff
@@ -231,18 +231,30 @@ export class MainStore {
...
     @action setLeftNavIndex(index) {
-        this.leftNavIndex = index;
+        this.leftMenuDrawer.set('index', index);
+    }
+    
+    @action toggleLeftMenuDrawer() {
+        this.leftMenuDrawer.set('open', !this.leftMenuDrawer.get('open'))
     }
 
-    @action toggleNavDrawer() {
```
Comment (caseychoiniere) [inline app/scripts/stores/mainStore.js:237]: this will break a test.

**Review by caseychoiniere — CHANGES_REQUESTED** [review]

**Review by caseychoiniere — APPROVED** [review]

**Pull request merged**